mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -336,6 +365,64 @@ def _try_transform_response_to_chat_format(response):
         else:
             return transformed_response.model_dump(mode="json")
 
+    @staticmethod
+    def _try_transform_response_iter_to_chat_format(chunk_iter):
+        from langchain_core.messages.ai import AIMessageChunk
+
+        def _gen_converted_chunk(message_content, finish_reason):
+            transformed_response = _ChatChunkResponse(
+                id=None,
+                created=int(time.time()),
+                model=None,
+                choices=[
+                    _ChatChoiceDelta(
+                        index=0,
+                        delta=_ChatDeltaMessage(
+                            role="assistant",
+                            content=message_content,
+                        ),
+                        finish_reason=finish_reason,
+                    )
+                ],
+            )
+
+            if Version(pydantic.__version__) < Version("2.0"):
+                return json.loads(transformed_response.json())
+            else:
+                return transformed_response.model_dump(mode="json")
+
+        def _convert(chunk):
+            if isinstance(chunk, str):
+                message_content = chunk
+                finish_reason = None
+                need_append_finish_chunk = True
+            elif isinstance(chunk, AIMessageChunk):
+                message_content = chunk.content
+                finish_reason = chunk.response_metadata.get("finish_reason")
+                need_append_finish_chunk = False
```

Comment:
what if `finish_reason` is None?

Reply:
the `_convert` is invoked in the iteration loop body. and the `need_append_finish_chunk` value returned in the last iteration is used in the following code. In the last iteration, the `finish_reason` is always not None.